### ISSUE TITLE
sass-loader Sourcemaps not preserved after passing through icon-font loader

### ISSUE THREAD

**Issue by knyy** [vusion/icon-font-loader#29]
Hello, Good Day,

I seem to be having issue with viewing the source maps generated from sass-loader when icon-font loader is used. When icon-font-loader is removed from loader chain, source maps for each sass file is able to be seen. Otherwise with icon-font-loader in the loader chain, we only see style.css sourcemap. Thank you very much for your time.

**Actual:**
Generated sourcemaps links to styles.css.

**Expected:**
Generated sourcemaps links to each individual js file used in the project e.g. my-component.scss, website-general.scss instead of styles.css

**Details:**
With the following loader sequence,  sass-loader -> icon-font-loader -> postcss -> css-loader, -> mini-css-extract. I think that the sass-loader sourcemap generated is not passed along the loader chain so we are not able to see the individual file. We only are able to see the style.css sourcemap generated via postcss loader.

I am using version 1.5.2 of icon-font-loader, webpack 4.

**Here is my loader sequence configuration**:

 {
    test: /\.(scss|css)$/,
    use: [
      process.env.NODE_ENV === "production" ? MiniCssExtractPlugin.loader : {
        loader: require.resolve("style-loader"),
        options: {
          sourceMap: true
        }
      },
      {
        loader: require.resolve("css-loader"),
        options: {
          sourceMap: true,
          importLoaders: 3
        }
      },
      {
        loader: require.resolve("postcss-loader"),
        options: {
          ident: "postcss",
          sourceMap: true,
          plugins: () => postCssPlugins
        }
      },
      {
        loader: require.resolve('icon-font-loader')
      },
      {
        loader: require.resolve("sass-loader"),
        options: {
          sourceMap: true,
          includePaths: [
            path.resolve(`${paths.root}/node_modules`)
          ]
        }
      }
    ]
  }, .....

**Comment by knyy** [comment]
Since by convention loaders pass the sourcemaps as the second parameter, icon-font-loader can preserve sourcemaps by modifying the loader to accept an additional param and return an additional param e.g.

function iconFontLoader(source, sourcemaps) {
    
    ...your code here
    
    callback(null, acceptPostCssAst ? ast : cssStr, sourcemaps);
}

**Comment by AlfredMou** [comment]
First of all, thank you for your help. We tried your method and this is valid. However, we found that our postcss processing will cause sourcemap confusion. Simply passing the sourcemap will not solve this problem. We need to regenerate the new sourcemap by passing the sourcemap to postcss. This will change the way we currently generate and process css. We need more time to change and test this feature, so this bug can't be fixed in a short time. Sorry, we didn't pay attention to the css sourcemap and caused these bugs, we will deal with them as soon as possible. Thank you again for your help。
